pkg/storage/unified/resource/kv.go

```diff
@@ -92,32 +92,71 @@ func (k *badgerKV) Get(ctx context.Context, section string, key string) (io.Read
 	return io.NopCloser(bytes.NewReader(value)), nil
 }
 
-func (k *badgerKV) Save(ctx context.Context, section string, key string, value io.Reader) error {
-	if k.db.IsClosed() {
-		return fmt.Errorf("database is closed")
+// badgerWriteCloser implements io.WriteCloser for badgerKV
+type badgerWriteCloser struct {
+	ctx     context.Context
+	db      *badger.DB
+	section string
+	key     string
```

Comment:
Nit: perhaps we could store only `keyWithSection`, we don't need to check for empty section again in the writer.